src/common/snippets/include/snippets/pass/tokenization.hpp

```diff
@@ -71,6 +71,9 @@ class SnippetsTokenization : public ov::pass::ModelPass {
               mha_token_enable_transpose_on_output(enable_transpose_on_output), mha_supported_transpose_ranks(std::move(mha_transpose_ranks)) {}
 
         size_t concurrency = 1;
+        // The number of grp that can be used as data pointers for data nodes (Parameter (and non-Scalar Constants),
+        // Result, Buffers with the same ID)
+        size_t data_ptr_grp_count = std::numeric_limits<size_t>::max();
```

Comment:
@IvanNovoselov  honestly, it was my [idea](https://github.com/openvinotoolkit/openvino/pull/23180#discussion_r1528015671) to use `SIZE_MAX` here to allow the pass to tokenize everything that it can. In my opinion, `SIZE_MAX` should be here instead of `zero` since the common pipeline shouldn't have any limitations related to data count on tokenization. I assume that it's default behavior since template plugin wouldn't have this limitation, snippets tests don't have this limitation.
If you don't want to tokenize Subgraphs, you shouldn't set `zero` here - you should just disable this pass 😃 
One more point: you don't need to update all unit snippets tests related to tokenization pass (in your case we should set new value in tokenization config) - the unit tests just validate common pipeline without limitations from backend side.

Reply:
We discussed the issue offline and decided that the best solution here is:
1) delete the default constructor for the config
2) update the non-default constructor, so it accepts all the fields as arguments (including data_ptr_gpr_count)
3) add arch-specific assert to the constructor that the provided number of registers can be supported on this architecture
4) move all the fields to to `private`, and implement appropriate getters
This way we can guarantee that all the values were set, and also can easily validate the provided arguments.

@xuchen-intel, do you think you can apply these changes in this PR? The description is long, but it can surely be done in no time 🙂

Reply:
Agree. Applied and set config manually for test cases accordingly. Thanks Ivan and Alexandra!